### ISSUE TITLE
UUID

### ISSUE THREAD

**Issue by jakearchibald** [whatwg/infra#153]
UUIDs seem to be the best way to express equality between instances of a concept, especially if that data lives between browser restarts.

Is this worth having in infra?

**Comment by annevk** [comment]
Yeah, it might be worth doing a survey to see where UUID is used as a primitive and what characteristics are assumed. And then define it and provide some guidance around it.

Service workers also expose it, right? And I think File API does too. Might be good to have serialization requirements too then.

**Comment by jakearchibald** [comment]
Service worker uses https://html.spec.whatwg.org/multipage/webappapis.html#concept-environment-id and exposes it via https://w3c.github.io/ServiceWorker/#dom-client-id.

I'm looking to use it in background fetch to distinguish instances of https://wicg.github.io/background-fetch/#background-fetch-concept, but it won't be exposed.

**Comment by annevk** [comment]
https://w3c.github.io/FileAPI/#ABNFUUID has a whole ABNF. I wonder if that's actually implemented as such though.

**Comment by annevk** [comment]
https://github.com/mozilla/standards-positions/issues/511#issuecomment-823443317 by @bcoe lists some more places. Perhaps that specification can be the central point of reference though.

**Comment by jakearchibald** [comment]
Related: In the session history PR I'm introducing the idea of a unique internal value https://whatpr.org/html/6315/common-microsyntaxes.html#unique-values
